{
  "analysis": {
    "error": null,
    "issues": [
      {
        "address": 260,
        "code": "function Command(address adr,bytes data)\n    payable\n    public\n    {\n        require(msg.sender == Owner);\n        // <yes> <report> UNCHECKED_LL_CALLS\n        adr.call.value(msg.value)(data);\n    }",
        "debug": "calldata_Pie_32 + 4: 0xfffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffc\ncalldata_Pie_0: 0xba21d62a00000000000000000000000000000000000000000000000000000000\ncalldatasize_Pie: 0x4\n",
        "description": "A possible integer overflow exists in the function `Command(address,bytes)`.\nThe addition or multiplication may result in a value higher than the maximum representable integer.",
        "filename": "/data/0xd2018bfaa266a9ec0a1a84b061640faa009def76.sol",
        "function": "Command(address,bytes)",
        "lineno": 38,
        "title": "Integer Overflow ",
        "type": "Warning"
      },
      {
        "address": 683,
        "code": "Owner.transfer(this.balance)",
        "debug": "",
        "description": "A possible transaction order independence vulnerability exists in function withdraw(). The value or direction of the call statement is determined from a tainted storage location",
        "filename": "/data/0xd2018bfaa266a9ec0a1a84b061640faa009def76.sol",
        "function": "withdraw()",
        "lineno": 35,
        "title": "Transaction order dependence",
        "type": "Warning"
      },
      {
        "address": 828,
        "code": "Owner.transfer(this.balance)",
        "debug": "",
        "description": "Multiple sends exist in one transaction, try to isolate each external call into its own transaction. As external calls can fail accidentally or deliberately.\nConsecutive calls: \nCall at address: 922\n",
        "filename": "/data/0xd2018bfaa266a9ec0a1a84b061640faa009def76.sol",
        "function": "Get()",
        "lineno": 25,
        "title": "Multiple Calls",
        "type": "Information"
      },
      {
        "address": 828,
        "code": "Owner.transfer(this.balance)",
        "debug": "",
        "description": "A possible transaction order independence vulnerability exists in function Get(). The value or direction of the call statement is determined from a tainted storage location",
        "filename": "/data/0xd2018bfaa266a9ec0a1a84b061640faa009def76.sol",
        "function": "Get()",
        "lineno": 25,
        "title": "Transaction order dependence",
        "type": "Warning"
      },
      {
        "address": 922,
        "code": "msg.sender.transfer(this.balance)",
        "debug": "SOLVER OUTPUT:\ncallvalue: 0xde0b6b3a7640001\nretval_828: 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff\ncalldata_Pie_0: 0xb1976a0200000000000000000000000000000000000000000000000000000000\ncalldatasize_Pie: 0x4\n",
        "description": "In the function `Get()` a non-zero amount of Ether is sent to msg.sender.\nIt seems that this function can be called without restrictions.",
        "filename": "/data/0xd2018bfaa266a9ec0a1a84b061640faa009def76.sol",
        "function": "Get()",
        "lineno": 26,
        "title": "Ether send",
        "type": "Warning"
      },
      {
        "address": 1208,
        "code": "adr.call.value(msg.value)(data)",
        "debug": "",
        "description": "This contract executes a message call to an address provided as a function argument. Generally, it is not recommended to call user-supplied addresses using Solidity's call() construct. Note that attackers might leverage reentrancy attacks to exploit race conditions or manipulate this contract's state.",
        "filename": "/data/0xd2018bfaa266a9ec0a1a84b061640faa009def76.sol",
        "function": "Command(address,bytes)",
        "lineno": 44,
        "title": "Message call to external contract",
        "type": "Warning"
      },
      {
        "address": 1208,
        "code": "adr.call.value(msg.value)(data)",
        "debug": "",
        "description": "The return value of an external call is not checked. Note that execution continue even if the called contract throws.",
        "filename": "/data/0xd2018bfaa266a9ec0a1a84b061640faa009def76.sol",
        "function": "Command(address,bytes)",
        "lineno": 44,
        "title": "Unchecked CALL return value",
        "type": "Informational"
      }
    ],
    "success": true
  },
  "contract": "dataset/unchecked_low_level_calls/0xd2018bfaa266a9ec0a1a84b061640faa009def76.sol",
  "duration": 19.128282070159912,
  "end": 1654179727.4502969,
  "exit_code": 0,
  "start": 1654179708.3220148,
  "tool": "mythril"
}